integration_test/list_test.go

```diff
@@ -40,5 +40,41 @@ func TestKrewList(t *testing.T) {
 	if diff := cmp.Diff(eventualList, expected); diff != "" {
 		t.Fatalf("'list' output doesn't match:\n%s", diff)
 	}
-	// TODO(ahmetb): install multiple plugins and see if the output is sorted
+
+	test.Krew("uninstall", validPlugin).RunOrFail()
+	expected = []byte{'\n'}
+
+	uninstallList := test.Krew("list").RunOrFailOutput()
+	if diff := cmp.Diff(uninstallList, expected); diff != "" {
+		t.Fatalf("'list' output doesn't match:\n%s", diff)
+	}
+
+	test.Krew("install", validPlugin5, validPlugin4, validPlugin3, validPlugin2, validPlugin).RunOrFail()
```

Comment:
I think checking the sorting is not a good test-case for e2e tests:

* it requires installing several plugins, which is slow
* it can be tested independently with a unit test, if the production logic is refactored and the sorting is pulled out of the `cmd` package.

So I'd recommend to remove this.

Reply:
> it requires installing several plugins, which is slow

The plugins selected are all shell scripts to keep things quick. I'm not sure how performance sensitive the testing actually is, `initializeIndex()` re-expands the indexDir multiple times in testing. Maybe all of `intiailzeIndex()` should fall under `initIndexOnce.Do()`?

Reply:
TBC at the new PR